packages/account/src/providers/resource-cache.ts

```diff
@@ -0,0 +1,79 @@
+import { ErrorCode, FuelError } from '@fuel-ts/errors';
+import { hexlify } from '@fuel-ts/utils';
+
+import type { ExcludeResourcesOption } from './resource';
+
+interface CachedResource {
+  utxos: Set<string>;
+  messages: Set<string>;
+  timestamp: number;
+}
+
+const cache = new Map<string, CachedResource>();
+
+export class ResourceCache {
+  readonly ttl: number;
```

Comment:
```suggestion
export class ResourceCache {
  readonly ttl: number;
  private cache = new Map<string, CachedResource>();
```

If somebody were to instantiate two providers, the cache would be shared between them because this `Map` is shared between all `ResourceCache` instances. The test below fails as a result. I suggest you add it to `provider.test.ts`.
```ts
  it('cache is not shared between Predicate instances', async () => {
    using launched1 = await setupTestProviderAndWallets();
    using launched2 = await setupTestProviderAndWallets();

    const {
      wallets: [sender, receiver],
    } = launched1;

    await sender.transfer(receiver.address, 1000);

    expect(launched1.provider.cache?.getActiveData()).not.toMatchObject(
      launched2.provider.cache?.getActiveData() ?? {}
    );
  });
```

This shouldn't happen because `cache` [isn't](https://github.com/FuelLabs/fuels-ts/blob/9c3eb66e133489bf7eee2aa59de8b5c5745f9db5/packages/account/src/providers/provider.ts#L376) a static property on `Provider`.

To fix this, `cache` should be internalized into `ResourceCache`.

Reply:
@nedsalk This is intentional. A provider can be instantiated in different parts of an dApp application.

Reply:
I discussed this privately with @Torres-ssf and he'll create a new issue/discussion on this topic, listing the implications of making the cache fully static or making it fully instance-specific. For now we agreed that it's fine as it is.